machine/v1/types_alibabaprovider.go

```diff
@@ -111,18 +122,22 @@ type AlibabaCloudMachineProviderConfig struct {
 	// +optional
 	SystemDisk SystemDiskProperties `json:"systemDisk,omitempty"`
 
-	// VSwitch is a reference to the vswitch to use for this instance
+	// VSwitch is a reference to the vswitch to use for this instance.
+	// A reference holds either the vSwitch ID, the resource name, or the required tags to search.
+	// When more than one vSwitch is returned for a tag search, only the first vSwitch returned will be used.
 	// This parameter is required when you create an instance of the VPC type.
 	// You can call the DescribeVSwitches operation to query the created vSwitches.
-	VSwitch AlibabaResourceReference `json:"vSwitch,omitempty"`
+	VSwitch AlibabaResourceReference `json:"vSwitch"`
 
 	// RAMRoleName is the name of the instance Resource Access Management (RAM) role. This allows the instance to perform API calls as this specified RAM role.
 	// +optional
 	RAMRoleName string `json:"ramRoleName,omitempty"`
 
-	// ResourceGroupID is the unique ID of the resource group to which to assign the instance.
-	// +optional
-	ResourceGroupID string `json:"resourceGroupId,omitempty"`
+	// ResourceGroup references the resource group to which to assign the instance.
+	// A reference holds either the resource group ID, the resource name, or the required tags to search.
+	// When more than one resource group are returned for a search, an error will be produced and the Machine will not be created.
+	// Resource Groups do not support searching by tags.
+	ResourceGroup AlibabaResourceReference `json:"resourceGroup"`
```

Comment:
Based on some of the conversations I've seen on slack, I think this should be optional